src/credentials.js

```diff
@@ -122,11 +125,18 @@ class Credentials extends EventEmitter {
         this.emit('login');
         return Promise.resolve();
       } else {
-        // Parse captcha image if it exists
+        // Parse additional login status if it exists
         let captcha = body.match(/<img id="captchaimg"[\s\S]+?>/im);
+        let addDevice = body.match(/<form id="frmNIDLogin"[\s\S]+?loginAndDeviceAdd[\s\S]+?<\/form>/im);
+        let otp = body.match(/<form id="frmNIDLogin"[\s\S]+?"otp"[\s\S]+?<\/form>/im);
         log('Failed to log in');
-        return Promise.reject('Invalid username or password' +
-          (captcha ? ('\n' + captcha[0]) : ''));
+        return Promise.reject({
+          text: 'Failed to log in.',
+          captcha: captcha ? captcha[0] : null,
+          addDevice: addDevice ? addDevice[0] : null,
```

Comment:
Is there a reason to change `deviceAdd` to `addDevice`? Since there's no information for `options`, I have no idea what fields are required, but if you're trying to use raw `options`, I'm pretty sure the field name would be `deviceAdd` or something.
If you're willing to provide an abstraction, it's okay to change it.